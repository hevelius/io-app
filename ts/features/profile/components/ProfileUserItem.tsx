import React from "react";
import { StyleSheet, View } from "react-native";
import { H3 } from "../../../components/core/typography/H3";
import { H4 } from "../../../components/core/typography/H4";
import ItemSeparatorComponent from "../../../components/ItemSeparatorComponent";
import customVariables from "../../../theme/variables";

type Props = {
  title: string;
  subtitle: string;
  icon: React.ReactElement;
};

const styles = StyleSheet.create({
  itemRow: {
    flexDirection: "row",
    alignItems: "center",
    padding: 10
  },
  rightPadding: {
    paddingRight: customVariables.contentPadding
  }
});

/**
 * This component is used to render a single item in the profile screen.
 * It is composed by a left icon, a title and a subtitle.
 * @param props
 * @returns
 */
const ProfileUserItem = (props: Props) => (
  <View>
    <View style={styles.itemRow}>
      <View style={styles.rightPadding} testID={"ProfileUserItemIconTestID"}>
        {props.icon}
      </View>
      <View>
        <H3 testID={"ProfileUserItemTitleTestID"}>{props.title}</H3>
        <H4 testID={"ProfileUserItemSubTitleTestID"} weight={"Regular"}>
          {props.subtitle}
        </H4>
      </View>
    </View>
    <ItemSeparatorComponent noPadded={true} />
  </View>
);

export default ProfileUserItem;
