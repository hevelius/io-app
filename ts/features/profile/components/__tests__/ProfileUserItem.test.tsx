import React from "react";
import { render } from "@testing-library/react-native";
import ProfileUserItem from "../ProfileUserItem";
import NameSurnameIcon from "../../../../img/assistance/nameSurname.svg";

type Props = {
  title: string;
  subtitle: string;
  icon: React.ReactElement;
};

const props: Props = {
  title: "title",
  subtitle: "subtitle",
  icon: <NameSurnameIcon width={24} height={24} />
};

describe("Test ProfileUserItem", () => {
  it("should render correctly", () => {
    const component = renderComponent({ ...props });
    expect(component).toBeTruthy();
  });
  it("should match the snapshot", () => {
    const component = renderComponent({ ...props });
    expect(component).toMatchSnapshot();
  });
  it("should show correctly the title", () => {
    const component = renderComponent({ ...props });
    expect(component.getByTestId("ProfileUserItemTitleTestID")).toBeDefined();
    expect(component.getAllByText(props.title)).toBeDefined();
  });
  it("should show correctly the subtitle", () => {
    const component = renderComponent({ ...props });
    expect(
      component.getByTestId("ProfileUserItemSubTitleTestID")
    ).toBeDefined();
    expect(component.getAllByText(props.subtitle)).toBeDefined();
  });
  it("should show correctly the icon", () => {
    const component = renderComponent({ ...props });
    expect(component.getByTestId("ProfileUserItemIconTestID")).toBeTruthy();
  });
});

const renderComponent = ({ title, subtitle, icon }: Props) =>
  render(<ProfileUserItem title={title} subtitle={subtitle} icon={icon} />);
