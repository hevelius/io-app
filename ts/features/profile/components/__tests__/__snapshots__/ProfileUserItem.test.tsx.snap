// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ProfileUserItem should match the snapshot 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "padding": 10,
      }
    }
  >
    <View
      style={
        Object {
          "paddingRight": 24,
        }
      }
      testID="ProfileUserItemIconTestID"
    >
      <SvgMock
        height={24}
        width={24}
      />
    </View>
    <View>
      <Text
        color="bluegreyDark"
        font="TitilliumWeb"
        fontStyle={
          Object {
            "fontSize": 18,
          }
        }
        style={
          Array [
            Object {
              "fontSize": 18,
            },
            Object {
              "color": "#17324D",
              "fontFamily": "Titillium Web",
              "fontStyle": "normal",
              "fontWeight": "600",
            },
          ]
        }
        testID="ProfileUserItemTitleTestID"
        weight="SemiBold"
        weightColorFactory={[Function]}
      >
        title
      </Text>
      <Text
        color="bluegreyDark"
        font="TitilliumWeb"
        fontStyle={
          Object {
            "fontSize": 16,
          }
        }
        style={
          Array [
            Object {
              "fontSize": 16,
            },
            Object {
              "color": "#17324D",
              "fontFamily": "Titillium Web",
              "fontStyle": "normal",
              "fontWeight": "400",
            },
          ]
        }
        testID="ProfileUserItemSubTitleTestID"
        weight="Regular"
        weightColorFactory={[Function]}
      >
        subtitle
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {},
        Object {
          "backgroundColor": "#C9C9C9",
        },
        false,
        Object {
          "height": 0.5,
        },
      ]
    }
  />
</View>
`;
